SQL-562:  Readme update with more details regarding how to use the JDBC driver (#107)
mongodb/mongo-jdbc-driver: README.md

<img height="90" alt="MongoDB Atlas JDBC Driver" align="right" src="resources/media/MongoDBAtlasJDBC.png" />

> # MongoDB Atlas JDBC Driver

Reviewer: ```suggestion
# MongoDB Atlas SQL JDBC Driver
```

Author: Done. I also updated the next occurrences I found in the file.

> MongoDB Atlas JDBC provides SQL connectivity to [Atlas](https://www.mongodb.com/atlas) databases and datalakes for client applications developed in Java.

Reviewer: ```suggestion
The MongoDB Atlas SQL JDBC Driver provides SQL connectivity to [MongoDB Atlas](https://www.mongodb.com/atlas) for client applications developed in Java.
See the [Atlas SQL Documentation](https://www.mongodb.com/docs/datalake/admin/query-with-sql/) for more information.

```

Author: Done.

MongoDB Atlas JDBC is a JDBC Type 4 driver compatible with the JDBC 4.2 specification.

### Prerequisites
MongoDB Atlas JDBC driver requires Java 1.8 or higher.

## Getting the Driver

### Download and Install
You can download the precompiled driver (jar) from [Maven Central](https://search.maven.org/artifact/org.mongodb/mongodb-jdbc).

> ### Build From Source

Reviewer: suggestion: let's group all these build instructions under a "Contributing" or "development" heading alongside the integration testing instructions below

Author: Yes, good idea. Done!

To build and test the driver run the following commands from root dir.

#### To build the lean jar which does not include the dependencies
```
./gradlew clean build
```

#### To run demo locally
```
./gradlew clean :demo:run
```

#### To generate the source Jar
```
./gradlew clean :sourceJar  
```
You can find the generated jar in build/libs/

#### To generate the fat Jar which includes all the dependencies
```
./gradlew clean :shadowJar
```
You can find the generated jar in build/libs/

#### To generate the test Jar
```
./gradlew clean :testJar  
```
You can find the generated jar in build/libs/

#### To run the unit tests
```
./gradlew clean test
```

#### To fix lint problem
```
./gradlew spotlessApply
```

> ## Usage

Reviewer: suggestion: let's move "Prerequisites" and "Download & Install" under this "Usage" heading, and have it be the first thing in the README after the summary, so that users don't have to scroll through any build/development instructions to find the usage info

Author: Sounds good to me. Done 👍

- Supported options: [https://www.mongodb.com/docs/manual/reference/connection-string/#std-label-connections-connection-options](https://www.mongodb.com/docs/manual/reference/connection-string/#std-label-connections-connection-options)

##### Notes
- Special characters in the JDBC url have to be URL encoded.
- The driver can not connect to a mongod instance, only to MongoDB Atlas.

#### Connection Properties
All connection options can also be specified through a Properties object parameter instead of the being directly in the URL.
However, if an option is in both the URL and the Properties object, the connection will fail.

In addition to the standard MongoDB connection options there are a number of additional properties specific to the JDBC driver.  
These properties can only be specified using an additional Properties object parameter and not in the URL.

| Property                      | Type    | Required | Default | Description   |
| ----------------------------- | ------- | -------- | :-----: | ------------- |
| database                      | String  | Yes      | Null    | The name of the database used when querying |

> | loglevel                      | String  | No       | OFF     | The log level used for logging. Supported levels by increasing verbosity are 'OFF', 'SEVERE', 'WARNING', 'FINE', 'INFO' and 'FINER' |

Reviewer 2: ```suggestion
| loglevel                      | String  | No       | OFF     | The log level used for logging. Supported levels by increasing verbosity are 'OFF', 'SEVERE', 'WARNING', 'INFO', 'FINE' and 'FINER' |
```
`fine` is more verbose than `info`

> | logdir                        | String  | No       | Null    | The directory to use for log files. If not logging directory is specified, the logs are sent to the console |

Reviewer 2: ```suggestion
| logdir                        | String  | No       | Null    | The directory to use for log files. If no logging directory is specified, the logs are sent to the console |
```

```
         java.util.Properties p = new java.util.Properties();
         p.setProperty("user", "user");
         p.setProperty("password", "foo");
         p.setProperty("database", "test");
         p.setProperty("loglevel", Level.SEVERE.getName());
         System.out.println("Connecting to database test...");
         Connection conn = DriverManager.getConnection("mongodb://mydatalake-xxxx.a.query.mongodb.net/?ssl=true&authSource=admin", p);
```

## Development

### Build From Source
To build and test the driver run the following commands from root dir.

#### To build the lean jar which does not include the dependencies
```
./gradlew clean build
```
#### To run demo locally
```
./gradlew clean :demo:run
```
#### To generate the source Jar
```
./gradlew clean :sourceJar
```
You can find the generated jar in build/libs/  

#### To generate the fat Jar which includes all the dependencies
```
./gradlew clean :shadowJar
```
You can find the generated jar in build/libs/  

#### To generate the test Jar
```
./gradlew clean :testJar
```
You can find the generated jar in build/libs/  

#### To run the unit tests
```
./gradlew clean test
```

> #### To fix lint problem

Reviewer 2: ```suggestion
#### To fix lint problems
```
